src/media-gallery/media-gallery.js

```diff
@@ -158,6 +225,13 @@ component.controller = function MediaGalleryController() {
             vm.onViewAll();
         }
     };
-};
+
+    vm.setView = function setView(view) {
+        vm.view = view;
+        if (typeof vm.onSetView === 'function') {
+            vm.onSetView(view);
```

Comment:
this is using an expression binding (`&`) so to pass an argument, we have to use named parameters, for example

```
vm.onSetView({ view: view})
```